Add channel-list-item component, reskin channels-grid

jonboiser opened the pull request:
This adds a 'channel-list-item' component that is supposed to be used in different parts of the new import/export workflows. It is used initially to reskin the import/export landing page and will be used later for the import and export workflows. See tests for intended usage in other workflows.

Design: https://projects.invisionapp.com/share/Q9D1VQD67#/screens/254175259

![screen shot 2017-10-10 at 12 28 03 pm](https://user-images.githubusercontent.com/10248067/31406598-da17f48a-adb6-11e7-86b6-fcc2c3a5ba84.png)

christianmemije commented on kolibri/plugins/management/assets/src/device_management/views/manage-content-page/channel-list-item.vue:
```diff
@@ -0,0 +1,172 @@
+
+    <div class="thumbnail dtc">
+      <mat-svg
+        v-if="channel.thumbnail === ''"
```
Can just do `v-if="!thumbnailImg"`

christianmemije commented on kolibri/plugins/management/assets/src/device_management/views/manage-content-page/channel-list-item.vue:
```diff
@@ -0,0 +1,172 @@
+        return this.channel.thumbnail;
+      },
+      tasksInQueue() {
+        if (this.pageState.taskList === undefined) {
```
Can combine into
```js
if (this.pageState.taskList && this.pageState.taskList.length) {
  return true;
}
return false;
```

christianmemije commented on kolibri/plugins/management/assets/src/device_management/views/manage-content-page/channel-list-item.vue:
```diff
@@ -0,0 +1,172 @@
+      <mat-svg
+        v-if="channel.thumbnail === ''"
+        category="action"
+        name="view_module"
```
Should be using the [apps](https://material.io/icons/#ic_apps) material icon. It is slightly different. Mind also updating it in the Device navbar?

Also, would look nicer if you gave the square a `$core-grey` background color and made the icon bigger, similar to what is done in the cards under Learn.

Note, you can also just use the `<content-icon/>` component instead of the `<mat-svg/>`.

christianmemije commented:
Also, looking at the `channels-grid`, a linear progress might look better than the circular progress. 

![before](https://user-images.githubusercontent.com/7193975/31410168-ec13f790-adc2-11e7-82d3-b46a59f44a2f.gif)

![after](https://user-images.githubusercontent.com/7193975/31410171-ed18deda-adc2-11e7-842e-efc77fe7a1b4.gif)

christianmemije commented:
Oops.

jonboiser commented:
@christianmemije 

1. Changed the circular progress on channels-grid to linear
1. Made adjustments to the place holder icon for channels w/o thumbnails (though they still look funky when you make the viewport really narrow).
1. Redid some of the conditionals, per comments
1. Changed icon on top navigation

![screen shot 2017-10-10 at 3 42 11 pm](https://user-images.githubusercontent.com/10248067/31414190-9d1538c0-add1-11e7-8793-80f4c6e3f571.png)
